sdks/python/apache_beam/runners/direct/evaluation_context.py

```diff
@@ -67,8 +72,17 @@ class _SideInputsContainer(object):
   def __init__(self, views):
     self._lock = threading.Lock()
     self._views = {}
+    self._transform_to_views = collections.defaultdict(list)
+
     for view in views:
       self._views[view] = _SideInputView(view)
+      self._transform_to_views[view.pvalue.producer].append(view)
+
+  def __repr__(self):
+    views = ', '.join([
+        str(elm) for elm in self._views.values()
+        ] if self._views.values() else [])
```

Comment:
<!--new_thread; commit:2053e91ce5cf41b329a4bf8bdb48a06b79b3626b; resolved:0-->
Same as above for `_SideInputView.__repr__`.

Reply:
Done